### ISSUE TITLE
Visualize the field component

### ISSUE THREAD

**Issue by kevin0074** [fancompute/legume#52]
Hi,

I tried to visualize the y component of the E field in the example "Guided mode expansion with Autograd". But the field is not confined in the cavity. What should I do to get the correct field distribution?

![Figure 2023-05-30 152635](https://github.com/fancompute/legume/assets/75712003/9d7ef502-edda-4a2d-b275-dd9027c1192f)

Thanks in advance.

**Comment by momchil-flex** [comment]
Are you saying that you ran the exact same example and you are getting this result, or did you modify the example in any way?

**Comment by momchil-flex** [comment]
Judging from the title of your plot, it seems that you have modified the PhC parameters, and are only computing a subset of the eigenvalues. My guess is that this is just the wrong mode, a higher-order mode of the cavity which is probably Ez-polarized at the center of the slab. You should probably decrease your `eig_sigma` to be closer to the expected fundamental mode frequency. And even then, note that the fundamental mode does not necessarily need to be mode 0 - you should try to look at the profiles of all computed modes to orient yourself in the cavity mode structure.

**Comment by kevin0074** [comment]
Hi, Thank you for your reply.
I apologize for the figure I uploaded yesterday, it was modified somehow. So I re-run the code which is exactly the same as the one in the example, but the same problem remains. It is weird that the absolute value of the field is correct but the real part of the field is not well confined.
The only thing I have changed to get this figure is
`ax = legume.viz.field(gme, 'e', 0, Nx*Ny, z=dslab/2, component='y', val='re', N1=300, N2=200)`
![Figure 2023-05-31 093134](https://github.com/fancompute/legume/assets/75712003/1b9943e8-4ee2-4b9c-967f-0d51e703af01)
